components/Chat/Message/MessageReactions/MessageReactionsDrawer/MessageReactionsDrawer.tsx

```diff
@@ -0,0 +1,163 @@
+    <BottomSheetModal
+      onDismiss={handleDismiss}
+      ref={bottomSheetModalRef}
+      topInset={insets.top}
+      snapPoints={["50%", "100%"]}
+    >
+      <BottomSheetContentContainer>
+        <BottomSheetHeader title="Reactions" hasClose />
+
+        {/* Preview of all reactions with counts and filter buttons */}
+        <ScrollView
+          horizontal
+          showsHorizontalScrollIndicator={false}
+          style={{
+            paddingLeft: theme.spacing.lg,
+            flexDirection: "row",
+          }}
+          scrollEnabled
+        >
+          {/* "All" button to clear the filter */}
+          <TouchableHighlight
+            onPress={() => setFilterReactions(null)}
+            style={getChipStyle(null)}
+            underlayColor={theme.colors.border.subtle}
+            accessible
+            accessibilityRole="button"
+            accessibilityLabel="Show all reactions"
+          >
+            <HStack style={{ alignItems: "center" }}>
+              <Text style={getChipTextColor(null)}>
+                All {rolledUpReactions.totalCount}
+              </Text>
+            </HStack>
+          </TouchableHighlight>
+
+          {/* Buttons for each unique reaction type with counts */}
+          {rolledUpReactions.preview.map((reaction, index) => (
+            <TouchableHighlight
+              key={index}
+              onPress={() =>
+                setFilterReactions(
+                  reaction.content === filterReactions ? null : reaction.content
+                )
+              }
+              style={getChipStyle(reaction.content)}
+              underlayColor={theme.colors.border.subtle}
+              accessible
+              accessibilityRole="button"
+              accessibilityLabel={`${reaction.count} ${reaction.content} reactions`}
+            >
+              <Text style={getChipTextColor(reaction.content)}>
+                {reaction.content} {reaction.count}
+              </Text>
+            </TouchableHighlight>
+          ))}
+          <HStack style={{ width: theme.spacing.xxl }} />
+        </ScrollView>
+
+        {/* Detailed list of each reaction, sorted and filtered with all own reactions on top */}
+        <BottomSheetScrollView>
+          {rolledUpReactions.detailed
+            .filter(
+              (item) => !filterReactions || item.content === filterReactions
+            )
+            .map((item, idx) => (
+              <HStack
+                key={`${item.content}-${item.reactor.address}-${idx}`}
+                style={styles.reaction}
+              >
+                <Avatar
+                  size={theme.avatarSize.md}
+                  uri={item.reactor.avatar}
+                  name={item.reactor.userName}
+                />
+                <Text style={styles.userName}>{item.reactor.userName}</Text>
+                <Text style={styles.reactionContent}>{item.content}</Text>
+              </HStack>
+            ))}
+        </BottomSheetScrollView>
```

Comment:
curious how big the groups we will have but if we end up have LOTS of reactions we might need to use a FlashList here. but we can keep for later maybe.

Reply:
@thierryskoda I've upgraded to `FlashList` to account for the potential many reactions we'll need to display in the drawer :) 🍿  https://cln.sh/K4HgZc6swdw2skX2Z4tK